internal/service/transfer/server.go

```diff
@@ -224,7 +224,7 @@ func ResourceServer() *schema.Resource {
 			"security_policy_name": {
 				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      SecurityPolicyName2018_11,
+				Default:      SecurityPolicyName2024_01,
```

Comment:
As a beneficiary of this PR, changing the default value may result into changing already provisioned infrastructure.

With the release of `TransferSecurityPolicy-2024-01`, it seems AWS has changed their defaults[1] to be the most recent version. 

I interpret the Terraform provider versioning best-practices[2] as that this is a change in Default values and therefor could constitute a Major release, while adding the additional policies a Minor release.

I'm looking for guidance (from anyone) if this should be split up into separate PRs or that this can be handled accordingly. I'm also happy to contribute on this issue.

1: https://web.archive.org/web/20240105171707/https://docs.aws.amazon.com/transfer/latest/userguide/security-policies.html
2: https://developer.hashicorp.com/terraform/plugin/best-practices/versioning#versioning-specification

Reply:
@robertgv Thanks for your work here 👏.
@omarwit Yes, changing an attribute's default value will have to be done in a major version release as the change would cause `terraform plan` diffs to appear when none were show previously (and if the plan were applied, then the Transfer Server resource in AWS will have it's security policy updated).

Reply:
To get this merged for this week's release I will go ahead and remove the changes to the attribute default value from this PR -- please open another GitHub Issue for the default value change and we can add it to the [**v6.0.0** milestone](https://github.com/hashicorp/terraform-provider-aws/milestone/306).